### ISSUE TITLE
XGBRanker label size predict size not match

### ISSUE THREAD

**Issue by pearlli98** [dmlc/xgboost#6698]
Trying to fit a XGB ranker with the code 
`ranker.fit(sc.fit_transform(X), y, group = occurCount)`

- Sizes of X and y are (9322, 28) (9322,). 
- The group array is [2569 1191 1878 1830 1854]. It adds up to 9322.
- Both X and y have been sorted in ascending order based on y.  
- Using rank:pairwise objective function. 

The ranker.fit() function is producing the error 
`[21:24:23] ../src/objective/rank_obj.cu:761: Check failed: preds.Size() == info.labels_.Size() (46610 vs. 9322) : label size predict size not match`

**Comment by trivialfis** [comment]
Could you please provide a reproducible example?

**Comment by pearlli98** [comment]
issue resolved!

**Comment by trivialfis** [comment]
@pearlli98 Would you like to share what happened?  Maybe we can provide better checks in the future?

**Comment by pearlli98** [comment]
Yes when I set the "num_class" parameter, the XGB ranker wouldn't generate. When I don't set it, it runs fine.

**Comment by trivialfis** [comment]
Ah, got it.  Ranker is not doing classification.
